go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -581,27 +586,29 @@ func (npw *nodePortWatcher) DeleteEndpoints(ep *kapi.Endpoints) {
 	}
 }
 
-func (npw *nodePortWatcher) UpdateEndpoints(old *kapi.Endpoints, new *kapi.Endpoints) {
-	name := ktypes.NamespacedName{Namespace: old.Namespace, Name: old.Name}
+func getEndpointAddresses(endpointSlice *discovery.EndpointSlice) []string {
+	endpointsAddress := make([]string, 0)
+	for _, endpoint := range endpointSlice.Endpoints {
+		endpointsAddress = append(endpointsAddress, endpoint.Addresses...)
+	}
+	return endpointsAddress
+}
 
-	if reflect.DeepEqual(new.Subsets, old.Subsets) {
+func (npw *nodePortWatcher) UpdateEndpointSlice(oldEpSlice, newEpSlice *discovery.EndpointSlice) {
+	oldEpAddr := getEndpointAddresses(oldEpSlice)
+	newEpAddr := getEndpointAddresses(newEpSlice)
+	if reflect.DeepEqual(oldEpSlice.Ports, newEpSlice.Ports) &&
```

Comment:
I don't _think_ we care if the epslice ports change here... because there would have to be a corresponding svc.Port change to have any effect on the actual rules on the node.  All we care about in this handler is did our state of "having host endpoints" change.